### PR TITLE
physical/spanner: use separate client for updating locks

### PR DESCRIPTION
**Pull request by sethvargo**
We believe this mitigates an issue where a large influx of requests cause the leader to be unable to update the lock table (since it cannot grab a client from the pool or the client has no more open connections), which causes cascading failure.

### CODE REVIEW

**Comment by sethvargo** [comment]
Test output:

```text
$ GOOGLE_SPANNER_DATABASE=projects/vs-test-282716/instances/test/databases/test GOOGLE_SPANNER_TABLE=Vault GOOGLE_SPANNER_HA_TABLE=VaultHA go test -v -count=1 ./physical/spanner/...
=== RUN   TestHABackend
2020-07-08T13:02:52.553-0400 [DEBUG] configuring backend
2020-07-08T13:02:52.553-0400 [DEBUG] creating HA client
2020-07-08T13:02:52.554-0400 [DEBUG] configuration: database=projects/vs-test-282716/instances/test/databases/test table=Vault haEnabled=true haTable=VaultHA maxParallel=0
2020-07-08T13:02:52.554-0400 [DEBUG] creating client
2020-07-08T13:02:52.554-0400 [DEBUG] configuring backend
2020-07-08T13:02:52.554-0400 [DEBUG] creating HA client
2020-07-08T13:02:52.555-0400 [DEBUG] configuration: database=projects/vs-test-282716/instances/test/databases/test table=Vault haEnabled=true haTable=VaultHA maxParallel=0
2020-07-08T13:02:52.555-0400 [DEBUG] creating client
--- PASS: TestHABackend (11.07s)
=== RUN   TestBackend
2020-07-08T13:03:03.113-0400 [DEBUG] configuring backend
2020-07-08T13:03:03.113-0400 [DEBUG] configuration: database=projects/vs-test-282716/instances/test/databases/test table=Vault haEnabled=false haTable=VaultHA maxParallel=0
2020-07-08T13:03:03.113-0400 [DEBUG] creating client
--- PASS: TestBackend (1.54s)
PASS
ok  	github.com/hashicorp/vault/physical/spanner	12.819s
```

File: physical/spanner/spanner.go [hunk 1]
```diff
@@ -79,18 +79,26 @@ type Backend struct {
...
-	// client is the underlying API client for talking to spanner.
-	client *spanner.Client
+	// haClient is the API client. This is managed separately from the main client
+	// because a flood of requests should not block refreshing the TTLs on the
+	// lock.
+	//
+	// These values will be nil if haEnabled is false.
+	haClient *spanner.Client
 
 	// logger and permitPool are internal constructs.
```
Comment (calvn) [inline physical/spanner/spanner.go:100]: `permitPool` is still referenced here.

File: physical/spanner/spanner.go [hunk 2]
```diff
@@ -79,18 +79,26 @@ type Backend struct {
...
 	// haEnabled indicates if high availability is enabled. Default: true.
 	haEnabled bool
 
-	// client is the underlying API client for talking to spanner.
-	client *spanner.Client
+	// haClient is the API client. This is managed separately from the main client
+	// because a flood of requests should not block refreshing the TTLs on the
+	// lock.
+	//
+	// These values will be nil if haEnabled is false.
```
Comment (calvn) [inline physical/spanner/spanner.go:97]:
```suggestion
	// This value will be nil if haEnabled is false.
```

**Comment by sethvargo** [comment]
@calvn updated. PTAL

**Pull request merged**